### PR TITLE
Add docs for OTEL tracing

### PR DESCRIPTION
**Pull request by akira**

### CODE REVIEW

File: docs/observability/how_to_guides/tracing/trace_with_opentelemetry.mdx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+pip install opentelemetry-sdk
+pip install opentelemetry-exporter-otlp
+```
+
+### 1. Configure your environment
+
+Setup environment variables for the endpoint, substitute your specific values:
+
+```bash
+export OTEL_OTEL_EXPORTER_OTLP_HEADERS=https://api.smith.langchain.com/otel
```
Comment (angus-langchain) [inline docs/observability/how_to_guides/tracing/trace_with_opentelemetry.mdx:35]: Is this correct, why does it have OTEL twice?

Comment (angus-langchain) [inline docs/observability/how_to_guides/tracing/trace_with_opentelemetry.mdx:35]: Why is this headers and not an endpoint

File: docs/observability/how_to_guides/tracing/trace_with_opentelemetry.mdx [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+export OTEL_EXPORTER_OTLP_HEADERS="x-api-key=<your langsmith api key>"
+```
+
+If you'd like to log to a different project than default, you can add the LANGSMITH_PROJECT:
```
Comment (angus-langchain) [inline docs/observability/how_to_guides/tracing/trace_with_opentelemetry.mdx:39]:
```suggestion
# Optional: Specify a custom project name (default is "default")
export OTEL_EXPORTER_OTLP_HEADERS="x-api-key=<your langsmith api key>,LANGSMITH_PROJECT=my-project-name"
```

File: docs/observability/how_to_guides/tracing/trace_with_opentelemetry.mdx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+            span.set_attribute(f"gen_ai.prompt.{i}.role", str(message["role"]))
+
+        completion = client.chat.completions.create(
+            model=model,
+            messages=messages
+        )
+
+        span.set_attribute("gen_ai.response.model", completion.model)
+        span.set_attribute("gen_ai.completion.0.content", str(completion.choices[0].message.content))
+        span.set_attribute("gen_ai.completion.0.role", "asistant")
```
Comment (angus-langchain) [inline docs/observability/how_to_guides/tracing/trace_with_opentelemetry.mdx:98]: nit: assistant

File: docs/observability/how_to_guides/tracing/trace_with_opentelemetry.mdx (hunk 1 above)
Comment (akira) [reply docs/observability/how_to_guides/tracing/trace_with_opentelemetry.mdx:35]: Oh - this is a typo

Comment (akira) [reply docs/observability/how_to_guides/tracing/trace_with_opentelemetry.mdx]: Typo, not sure how that go in there :)

File: docs/observability/how_to_guides/tracing/trace_with_opentelemetry.mdx [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+import { ConfigureSDKEnvironmentCodeTabs } from "@site/src/components/QuickStart";
+import {
+  CodeTabs,
+  typescript,
+} from "@site/src/components/InstructionsWithCode";
+
+# Trace with OpenTelemetry
+
+LangSmith can now accept traces from from OpenTelemetry based clients.
```
Comment (agola11) [inline docs/observability/how_to_guides/tracing/trace_with_opentelemetry.mdx:13]:
```suggestion
LangSmith can accept traces from from OpenTelemetry based clients.
```

**Review by agola11 — APPROVED** [review]

**Pull request merged**